### PR TITLE
GEODE-9022: Support Info Command

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
- add info command to supported list
- add unit/integration tests
- add dunit concurrency test

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractInfoIntegrationTest.java [hunk 1]
```diff
@@ -134,7 +132,7 @@ public void shouldReturnRedisVersion() {
 
   @Test
   public void shouldReturnTCPPort() {
-    int expectedPort = getExposedPort();
+    int expectedPort = getPort();
     String expectedResult = "tcp_port:" + expectedPort;
```
Comment (sabbey37) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractInfoIntegrationTest.java:136]: I think this needs to be changed back to `getExposedPort()`

**Review by sabbey37 — CHANGES_REQUESTED** [review]

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/InfoDUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  private static Properties locatorProperties;
+
+  private static MemberVM locator;
+  private static MemberVM server1;
+  private static MemberVM server2;
+
+  private static int redisServerPort1;
+  private static int redisServerPort2;
+
+  private static final AtomicInteger numInfoCalled = new AtomicInteger();
```
Comment (sabbey37) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/InfoDUnitTest.java:60]: It looks like we increment this in the `getInfo` method, but never use this in the test. Could it be removed? Apologies for noticing this after the initial review.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java [hunk 3]
```diff
@@ -60,9 +61,20 @@
...
+  private static final String USED_MEMORY = "used_memory";
+  private static final String MEMORY_FRAGMENTATION = "mem_fragmentation_ratio";
+  private static final String TCP_PORT = "tcp_port";
+  private static final String MAX_MEMORY = "maxmemory";
 
   private static final AtomicInteger numInfoCalled = new AtomicInteger(0);
 
+  public int getExposedPort() {
+    return -1; // This must be overridden by implementation classes
+  }
```
Comment (sabbey37) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java:73]: I like the way we used to do this in `AbstractInfoIntegrationTest` (declaring the method abstract so it had to be overridden):
```
abstract int getExposedPort();
```

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java [hunk 4]
```diff
@@ -60,9 +61,20 @@
...
+
+  public void configureMaxMemory(Jedis jedis) throws Exception {
+    throw new Exception("configureMaxMemory must be overridden");
+  }
```
Comment (sabbey37) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java:77]: Maybe we could declare `configureMaxMemory` abstract as well, like `getExposedPort`

**Review by sabbey37 — CHANGES_REQUESTED** [review]

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/InfoDUnitTest.java [hunk 5]
```diff
@@ -116,7 +114,7 @@ public void testConcurrentInfoFromDifferentServers_doNotStepOnEachOther() {
           int commandsProcessed2 = Integer.valueOf(info2.get(COMMANDS_PROCESSED));
           assertThat(commandsProcessed2).isGreaterThanOrEqualTo(previousCommandsProcessed2.get());
           previousCommandsProcessed2.set(commandsProcessed2);
-        }).runInLockstep();
+        }).run();
```
Comment (sabbey37) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/InfoDUnitTest.java:117]: Just wanted to make sure this was intentionally changed.

**Review by sabbey37 — APPROVED** [review]

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/InfoDUnitTest.java (hunk 5 above)
Comment (nonbinaryprogrammer) [reply geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/InfoDUnitTest.java:117]: yes, it was, thanks for checking

**Pull request merged**